infrastructure/src/main/java/org/corfudb/infrastructure/log/InMemoryStreamLog.java

```diff
@@ -34,6 +35,15 @@ public InMemoryStreamLog() {
         startingAddress = 0;
     }
 
+    @Override
+    public synchronized void append(List<LogData> entries) {
+        for (LogData entry : entries) {
+            if (isTrimmed(entry.getGlobalAddress()) || logCache.containsKey(entry.getGlobalAddress())) {
+                continue;
```

Comment:
We are missing the append of non-trimmed/non-cached entries.

Reply:
Yes. I'll add tests too.

Reply:
Done.

Reply:
done